entrypoint.sh

```diff
@@ -407,12 +413,38 @@ runPostSetupScripts() {
     set -e
     echo "Post setup scripts execution succeeded."
 }
+function runCertbotAsNeeded() {
+    if [ ! "$GENERATE_CERTBOT_CERT_SCHEDULED" = "True" ]; then
+      echo "No scheduled cert creation."
+      return
+    fi
+
+    echo "Waiting for Zulip to come online to generate cert ..."
+    while ! curl -sk "$SETTING_EXTERNAL_HOST" >/dev/null 2>&1; do
+      sleep 1;
+    done
+
+    echo "Generating LetsEncrypt cert ..."
+
+    # Remove the self-signed certs which were only needed to get Zulip going.
+    rm -f "$DATA_DIR"/certs/zulip.key "$DATA_DIR"/certs/zulip.combined-chain.crt
+
+    # Accept the terms of service automatically.
+    echo 'a' | \
+      /home/zulip/deployments/current/scripts/setup/setup-certbot \
+      --hostname="$SETTING_EXTERNAL_HOST" \
+      --email="$SETTING_ZULIP_ADMINISTRATOR" \
+      --deploy-hook "$ZULIP_CERTBOT_DEPLOY_HOOK"
```

Comment:
I'm not going to block on this, but it would be worth adding a `--accept-tos` option to `scripts/setup/setup-certbot` in zulip/zulip that we can use to clean this `echo` logic up.

Reply:
I did. :) It doesn't seem to work very reliably though.

https://github.com/zulip/zulip/pull/10015/files#diff-1d7a8d22d9800f5bc0462b04cf7235f1R94

Reply:
You just need to update the Git ref to point to your fork and then rebuild the Docker image to be able to use it, since that code is in the Docker image.

Reply:
Remember, I have a production setup with this all working. That wouldn't be possible without my pointing at the right git fork to get the deploy hooks I added. Still, with `--agree-tos`, I needed the echo.

Reply:
I'll try it again, though, since I was in the trenches last time and something else could've been awry.

Reply:
Sweet. I'm wrong! This is working just fine still.